spark/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -135,15 +135,22 @@ private[delta] class ConflictChecker(
    * the transaction as if it had started while reading the `winningCommitVersion`.
    */
   def checkConflicts(): CurrentTransactionInfo = {
+    // Protocol and metadata checks.
     checkProtocolCompatibility()
     checkNoMetadataUpdates()
-    checkForAddedFilesThatShouldHaveBeenReadByCurrentTxn()
-    checkForDeletedFilesAgainstCurrentTxnReadFiles()
-    checkForDeletedFilesAgainstCurrentTxnDeletedFiles()
+    checkIfDomainMetadataConflict()
     checkForUpdatedApplicationTransactionIdsThatCurrentTxnDependsOn()
+
+    // Row Tracking reconciliation. We perform this before the file checks to ensure that
+    // no files have duplicate row IDs.
```

Comment:
(and enforce the delta protocol/ align the files actions with the delta protocol/ avoid interacting with files that don't comply with the protocol)

Reply:
done.